packages/sdks/web-component/src/lib/descope-wc/sdkScripts/grecaptcha.ts

```diff
@@ -0,0 +1,119 @@
+declare global {
+  interface Window {
+    grecaptcha?: {
+      enterprise?: any;
+      ready: (callback: () => void) => void;
+      execute: (widgetId: any, options: { action: string }) => Promise<string>;
+      render: (container: HTMLElement, parameters: any) => any;
+    };
+    onRecaptchaLoadCallback: () => void;
+  }
+}
+
+export const loadGRecaptcha = (
+  initArgs: {
+    enterprise: boolean;
+    siteKey: string;
+  },
+  _inputs: { baseUrl?: string },
+  onTokenReady: (token: string) => void,
+) => {
+  const getScriptURL = () => {
+    const url = new URL('https://www.google.com/recaptcha/');
+    url.pathname += `${initArgs.enterprise ? 'enterprise' : 'api'}.js`;
+    url.searchParams.append('onload', 'onRecaptchaLoadCallback');
+    url.searchParams.append('render', 'explicit');
+    return url.toString();
+  };
+
+  const loadRecaptchaScript = () => {
+    const script = document.createElement('script');
+    script.src = getScriptURL();
+    script.async = true;
+    script.id = 'recaptcha-script';
+    script.defer = true;
+    document.body.appendChild(script);
+  };
+
+  const getGrecaptchaInstance = () =>
+    initArgs.enterprise ? window.grecaptcha?.enterprise : window.grecaptcha;
+
+  let timer;
+  let recaptchaWidgetId;
+
+  const getNewToken = (grecaptchaInstance, currentNode) => {
+    grecaptchaInstance.ready(() => {
+      if (!initArgs.siteKey) {
+        return;
+      }
+      grecaptchaInstance
+        ?.execute(recaptchaWidgetId, { action: 'load' })
+        .then((token: string, e: any) => {
+          if (e) {
+            // eslint-disable-next-line no-console
+            console.warn('could not execute recaptcha', e);
+          } else {
+            onTokenReady(token);
+            // if the component is still connected, we should try to get a new token before the token expires (2 minutes)
+            timer = setTimeout(() => {
+              getNewToken(grecaptchaInstance, currentNode);
```

Comment:
should we add a debug log when token is re-invoked ?

Reply:
Not sure, can't get spammy